app/Libraries/Search/BeatmapsetSearch.php

```diff
@@ -398,6 +405,17 @@ private function addTextFilter(BoolQuery $query, string $paramField, array $fiel
         $query->must($subQuery);
     }
 
+    private function addTagsFilter(BoolQuery $query): void
+    {
+        if (!present($this->params->tags)) {
+            return;
+        }
+
+        foreach ($this->params->tags as $tag) {
+            $query->filter(QueryHelper::queryString($tag, ['beatmaps.top_tags'], 'and'));
+        }
```

Comment:
This filter means that specifying multiple tags requires the beatmapset to have a beatmap that has all the tags; having the tags across different beatmaps in the set will not match.

Reply:
that'd be weird anyway. just like specifying multiple different difficulty parameters should match beatmap which has all those parameters, not when they're spread across multiple beatmaps within same beatmapset.